rasa/core/policies/embedding_policy.py

```diff
@@ -435,199 +232,117 @@ def train(
-
-        if self.evaluate_on_num_examples:
-            session_data, eval_session_data = train_utils.train_val_split(
-                session_data,
-                self.evaluate_on_num_examples,
-                self.random_seed,
-                label_key="action_ids",
-            )
-        else:
-            eval_session_data = None
-
-        self.graph = tf.Graph()
-        with self.graph.as_default():
-            # set random seed in tf
-            tf.set_random_seed(self.random_seed)
-
-            # allows increasing batch size
-            batch_size_in = tf.placeholder(tf.int64)
-
-            (
-                self._iterator,
-                train_init_op,
-                eval_init_op,
-            ) = train_utils.create_iterator_init_datasets(
-                session_data,
-                eval_session_data,
-                batch_size_in,
-                self.batch_strategy,
-                label_key="action_ids",
-            )
+        self.config[NUM_NEG] = min(self.config[NUM_NEG], domain.num_actions - 1)
 
-            self._is_training = tf.placeholder_with_default(False, shape=())
-
-            loss, acc = self._build_tf_train_graph()
-
-            # define which optimizer to use
-            self._train_op = tf.train.AdamOptimizer().minimize(loss)
-
-            # train tensorflow graph
-            self.session = tf.Session(config=self._tf_config)
-            train_utils.train_tf_dataset(
-                train_init_op,
-                eval_init_op,
-                batch_size_in,
-                TrainingMetrics(loss={"loss": loss}, score={"acc": acc}),
-                self._train_op,
-                self.session,
-                self._is_training,
-                self.epochs,
-                self.batch_size,
-                self.evaluate_on_num_examples,
-                self.evaluate_every_num_epochs,
-            )
+        # extract actual training data to feed to model
+        model_data = self._create_model_data(training_data.X, training_data.y)
 
-            # rebuild the graph for prediction
-            self.pred_confidence = self._build_tf_pred_graph(session_data)
+        # keep one example for persisting and loading
+        self.data_example = {k: [v[:1] for v in vs] for k, vs in model_data.items()}
 
-            self.attention_weights = train_utils.extract_attention(
-                self.attention_weights
-            )
+        self.model = TED(
+            self.config,
+            isinstance(self.featurizer, MaxHistoryTrackerFeaturizer),
+            self._encoded_all_label_ids,
+        )
+
+        self.model.fit(
+            model_data,
+            self.config[EPOCHS],
+            self.config[BATCH_SIZES],
+            self.config[EVAL_NUM_EXAMPLES],
+            self.config[EVAL_NUM_EPOCHS],
+            batch_strategy=self.config[BATCH_STRATEGY],
+            random_seed=self.config[RANDOM_SEED],
+        )
 
     def continue_training(
```

Comment:
I guess it will be removed in the 2.0 release. The function is still called, just a warning is printed.
See https://github.com/RasaHQ/rasa/blob/master/rasa/core/agent.py#L596.